receiver_test.go

```diff
@@ -0,0 +1,128 @@
+package amqp
```

Comment:
These tests could be simplified if the codebase started to decouple the different concrete implementations with interfaces.
I was able to work around the asynchronous and concurrent behavior in other ways, but they complicate the test code.